edb/server/dbview/dbview.pyx

```diff
@@ -966,9 +951,17 @@ cdef class DatabaseConnectionView:
                     # discard cache entry that cannot be recompiled
                     self._db._eql_to_compiled.pop(query_req, None)
                 else:
-                    await compiled_queue.put(
-                        (query_req, result[0], schema_version)
-                    )
+                    # schema_version, database_config and system_config are not
+                    # serialized but they affect the cache key. We only update
+                    # these values *after* the compilation because 1) they are
+                    # not needed for compilation, and 2) we can evict the in-
+                    # memory cache by the same key when recompilation fails.
```

Comment:
`database_config` and `system_config` can affect the compilation, though. Am I missing something?

Reply:
user_schema, database_config and system_config are not serialized in the CompilationRequest because they are sent to the compiler process separately, while they are all affecting the cache key because they affect the compilation. I think I shouldn't say "they are not needed for compilation", but rather like "the compiler takes their values separately"

Reply:
I tweaked the comments, and fixed a potential race issue on db_config and sys_config